lib/eventData.ts

```diff
@@ -77,51 +237,53 @@ export namespace EventData {
     }
     return data;
   }
-
+  /**
+   * Converts an EventData object to an AMQP message.
+   * @param {EventData} data The EventData object that needs to be converted to an AMQP message.
+   */
   export function toAmqpMessage(data: EventData): AmqpMessage {
     const msg: AmqpMessage = {
-      body: data.body
+      body: data.body,
+      // As per the AMQP 1.0 spec If the message-annotations or delivery-annotations section is omitted,
+      // it is equivalent to a message-annotations section containing anempty map of annotations.
+      message_annotations: {},
```

Comment:
you should be able to do `{} as AmqpMessageAnnotations` to avoid all of the `(msg.message_annotations as AmqpMessageAnnotations)` below (though you may not need either of them)

Reply:
same for the other ones

Reply:
Interesting never thought about that. I was like I am initializing it then why is TS complaining.